ext/drx/drx.c

```diff
@@ -2791,6 +2825,21 @@ skip_unknown_instr_inc(int reset_state, drx_state_machine_params_t *params)
     }
 }
 
+static void
+restore_spilled_xmm_value(drx_state_machine_params_t *params)
+{
+    byte xmm_val[XMM_REG_SIZE];
+    ASSERT(params->spilled_xmm_slot_addr != NULL,
+           "No spill address recorded for the app xmm value");
+    ASSERT(params->spilled_xmm != DR_REG_NULL && reg_is_strictly_xmm(params->spilled_xmm),
+           "No spilled xmm reg recorded");
+    void *spilled_xmm_val = params->spilled_xmm_slot_addr;
+    for (int i = 0; i < XMM_REG_SIZE; i++) {
+        xmm_val[i] = *((byte *)spilled_xmm_val + i);
```

Comment:
Doing dwords at a time or using memcpy should be faster

Reply:
Done.